components/sidebar/ObserverSection.js

```diff
@@ -0,0 +1,71 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import 'intersection-observer';
+
+const ObserverSection = ({ id, rootElement, rootMargin, granularity, index, setIntersectionRatios, children }) => {
+    // the granularity prop should be an integer and is meant to mark how many divisions we make to
+    // an area in order to observe it with an intersectionObserver.
+    //
+    // E.g. granularity = 2 will mean that intersectionObserver will emit events when
+    // the 0%, 50% or 100% of the area overlaps with a given target area.
+    //
+    // In general, the higher the granularity, the more sensitive we will be to changes in overlapping areas.
+    // Probably granularity = 20 will suffice for the purpose of updating URL in function of the dislpay area
+
+    const buildThresholds = (granularity) => {
+        const steps = parseInt(granularity, 10);
+        const thresholds = [];
+
+        for (let i = 0; i <= steps; i++) {
+            thresholds.push(i / steps);
+        }
+        return thresholds;
+    };
+
+    const handleIntersect = (entries) => {
```

Comment:
Could we debounce this function to 500ms to avoid overcharging on his parent?

Reply:
Done!